recipes/cli11/all/conanfile.py

```diff
@@ -1,45 +1,64 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.files import get, copy, rmdir
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.build import check_min_cppstd
 import os
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.52.0"
 
 class CLI11Conan(ConanFile):
     name = "cli11"
     homepage = "https://github.com/CLIUtils/CLI11"
     description = "A command line parser for C++11 and beyond."
-    topics = ("cli-parser", "cpp11", "no-dependencies", "cli", "header-only")
+    topics = "cli-parser", "cpp11", "no-dependencies", "cli", "header-only"
     url = "https://github.com/conan-io/conan-center-index"
     license = "BSD-3-Clause"
     settings = "os", "compiler", "build_type", "arch"
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "11"
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CLI11_BUILD_EXAMPLES"] = False
+        tc.variables["CLI11_BUILD_TESTS"] = False
+        tc.variables["CLI11_BUILD_DOCS"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
         cmake = CMake(self)
-        cmake.definitions["CLI11_BUILD_EXAMPLES"] = False
-        cmake.definitions["CLI11_BUILD_TESTS"] = False
-        cmake.definitions["CLI11_BUILD_DOCS"] = False
-        cmake.configure(source_folder=self._source_subfolder)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         # since 2.1.1
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()   # header only
```

Comment:
```suggestion
        self.info.clear()
```
Please, don't need comment here, it's documented already, and it's expected that people read the documentation.